### PR TITLE
Migrate projects back to csproj

### PR DESCRIPTION
**Pull request by xt0rted**
The json/xproj projects are no longer supported so in order to work on this project in the latest VS2017 we need to update to the new msbuild/csproj system.

Key changes in this PR
- Updates the project files to the new msbuild/csproj format
- Updates the travis config with some dotnet cli hacks (ref: https://github.com/dotnet/sdk/issues/335)
- Removes stand-alone sample project from main solution
- Drops the full framework version down to 4.5 (ref: #471)
- Updates sample project dependencies

By updating to the new project system you'll now need VS2017 in order to work on the project. This should also allow you to use VS for Mac though that's untested at the moment.

There's nothing in the repo around creating the NuGet package so the process for doing that may need to be updated. You can run `dotnet pack` to create the `.nupkg` so if that's how it's been done then nothing changes there.

This is a precursor to looking at #441 which is going to be effecting me in the coming days as I start to update my dependencies to their latest versions.

Closes #471

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-csharp?pullRequest=471) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-csharp?pullRequest=471) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-csharp?pullRequest=471) it.</sub>

**Comment by thinkingserious** [comment]
Thank you for working on this @xt0rted!

I hope these changes will allow us to work on the library in Visual Studio for the Mac too :)

**Comment by xt0rted** [comment]
@thinkingserious I'm not sure what the deal is with build failing on mono. This should work but I can't figure out what's wrong with the setup that's causing it to fail.

**Comment by xt0rted** [comment]
🎉 Hooray for dotnet cli hacks!

**Comment by xt0rted** [comment]
Since I was in here changing this around I dropped the full framework version back down to 4.5. I know this isn't supported anymore but there's been a couple requests for it recently and it doesn't really hurt since we can target that without any modifications to the code right now. I can split that out into a separate PR if you'd like.

**Comment by xt0rted** [comment]
I think this should be good for review now. It does update some of the dependencies but I don't think that's a breaking change. Adding .net 4.5 support back might be breaking but I'm not entirely sure about that.

**Comment by xt0rted** [comment]
If we want to keep it at 4.5.2 I can just rebase and drop that commit.

File: src/SendGrid/SendGrid.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <VersionPrefix>9.4.1</VersionPrefix>
-    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
+    <TargetFrameworks>netstandard1.3;net452</TargetFrameworks>
```
Comment (xt0rted) [inline src/SendGrid/SendGrid.csproj]: If we want to keep it at 4.5.2 I can just rebase and drop that commit.

**Comment by SendGridDX** [comment]
@xt0rted,

Does the project load and build for you on Mac OSX visual studio?

**Comment by SendGridDX** [comment]
For me, I get a bunch of errors (80)

**Comment by xt0rted** [comment]
I haven't had a chance to test this out on macOS. I won't have a chance to do that until tomorrow most likely.

**Comment by thinkingserious** [comment]
No worries, that would just be a bonus at this point :)

I'll be merging this one shortly.

**Comment by xt0rted** [comment]
I added OSX to the build matrix to see what that gives us, those builds take a lot longer to start though.

**Comment by thinkingserious** [comment]
@xt0rted,

Yes, it's been 20 minutes already. Please remove that test and then I'll go ahead and merge this. We can setup a different ticket for Mac OSX Visual Studio support.

**Comment by thinkingserious** [comment]
Hello @xt0rted,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by thinkingserious** [comment]
Hello @xt0rted,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**